### PR TITLE
support bucket level throttling

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
support bucket level throttling

## Motivation and Context
allow for bandwidth control over
I/O from buckets applies to both
PUT/GETs for now.

## How to test this PR?
Needs https://github.com/minio/mc/pull/4678

```
mc quota set alias/bucket --rate 1g
```

```
mc cp ~/Downloads/ubuntu-22.04.2-desktop-amd64.iso myminio/testbucket/
.../ubuntu-22.04.2-desktop-amd64.iso: 1.64 GiB / 4.59 GiB ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉┉━ 121.27 MiB/s
```

```
mc quota clear alias/bucket
```

```
mc cp ~/Downloads/ubuntu-22.04.2-desktop-amd64.iso myminio/testbucket/
.../ubuntu-22.04.2-desktop-amd64.iso: 4.59 GiB / 4.59 GiB ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 695.71 MiB/s 6s
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by klauspost — COMMENTED** [review]
Driveby review.

Is there some cross-server communication or is this a per server limits?

File: docs/bucket/throttling/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Bucket Throttling Guide [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io) [![Docker Pulls](https://img.shields.io/docker/pulls/minio/minio.svg?maxAge=604800)](https://hub.docker.com/r/minio/minio/)
+
+MinIO bucket level throttling allows for buckets to be configured with 
+
+- a bandwidth cap for both ingress and egress. Provides a way to provision bucket for customers with expected SLAs.
+- a number of requests per second. Provides a way to provision bucket for customers with expected SLAs.
```
Comment (klauspost) [inline docs/bucket/throttling/README.md:6]: It presume the reqs/s is implemented elsewhere, since I don't see it in the code.

Comment (harshavardhana) [reply docs/bucket/throttling/README.md:6]: Not yet @klauspost not going to be in this PR.

Comment (harshavardhana) [reply docs/bucket/throttling/README.md:6]: Docs I am still working on so it's not yet updated.

**Comment by harshavardhana** [comment]
> Driveby review.
> 
> Is there some cross-server communication or is this a per server limits?

It is a per server limit - set a value and divide that by node count.

**Comment by harshavardhana** [comment]
Actually there has been change of plans. Moving this out of MinIO.

**Pull request closed without merging**